### PR TITLE
Companies overhaul

### PR DESCRIPTION
**Pull request by LudvigHz**

### CODE REVIEW

File: lego/apps/companies/views.py [hunk 1]
```diff
@@ -31,6 +31,7 @@ class CompanyViewSet(
 ):
     queryset = Company.objects.all()
     permission_classes = (permissions.IsAuthenticated, )
+    ordering = 'name'
```
Comment (odinuge) [inline lego/apps/companies/views.py]: Smart :+1:

File: lego/apps/companies/serializers.py [hunk 2]
```diff
@@ -79,10 +80,21 @@ class CompanyListSerializer(BasisModelSerializer):
         }
     )
 
+    event_count = serializers.SerializerMethodField()
+
+    def get_event_count(self, obj):
```
Comment (odinuge) [inline lego/apps/companies/serializers.py]: We generally have all the class values before functions in our serielizer, so the line `joblisting_count = serializers.SerializerMethodField()` should be over this function :smile:

File: lego/apps/companies/serializers.py [hunk 3]
```diff
@@ -79,10 +80,21 @@ class CompanyListSerializer(BasisModelSerializer):
...
+        return obj.events.filter(start_time__gt=timezone.now()).count()
+
+    joblisting_count = serializers.SerializerMethodField()
+
+    def get_joblisting_count(self, obj):
```
Comment (odinuge) [inline lego/apps/companies/serializers.py]: Have you checked how many `sql`-queries these two method fields generates? I suspect they will create two queries per company, and that would result in `1 + (2 * companyCount)` queries (if all the prefetching is done in the view). :thinking:

File: lego/apps/companies/serializers.py (hunk 2 above)
Comment (LudvigHz) [reply lego/apps/companies/serializers.py]: got it! Will fix

File: lego/apps/companies/serializers.py (hunk 3 above)
Comment (LudvigHz) [reply lego/apps/companies/serializers.py]: As far as I can tell, they only generate 1 query each, not sure how to properly check this though.

**Pull request merged**